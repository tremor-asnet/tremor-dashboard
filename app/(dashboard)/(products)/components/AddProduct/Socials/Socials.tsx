// Libs
import { Controller, FieldErrors } from "react-hook-form";

// Components
import InputField from "@/components/common/CustomField/InputField/InputField";

// Constants
import { MESSAGES_ERROR, REGEX } from "@/constants";

// Types
import { NewSocial } from "@/types";

interface SocialsProps {
  control: any;
  errors: FieldErrors<NewSocial>;
}

const Socials = ({ control, errors }: SocialsProps) => {
  const facebookErrorMsg = errors.facebookUrl?.message;
  const instagramErrorMsg = errors.instagramUrl?.message;
  const shopifyErrorMsg = errors.shopifyUrl?.message;

  return (
    <div className="flex flex-col gap-6">
      <Controller
        name="shopifyUrl"
        control={control}
        rules={{
          required: MESSAGES_ERROR.URL_REQUIRED,
          pattern: {
            value: REGEX.URL,
            message: MESSAGES_ERROR.URL_REQUIRED,
          },
        }}
        render={({ field }) => (
          <>
            <InputField
              label="Shopify Handle"
              placeholder=""
              type="text"
              {...field}
            />
            <p className="text-[11px] xs:text-xs text-red-500">
              {shopifyErrorMsg}
            </p>
          </>
        )}
      />

      <Controller
        name="facebookUrl"
        control={control}
        rules={{
          required: MESSAGES_ERROR.URL_REQUIRED,
          pattern: {
            value: REGEX.URL,
            message: MESSAGES_ERROR.URL_REQUIRED,
          },
        }}
        render={({ field }) => (
          <>
            <InputField
              label="Facebook Account"
              placeholder=""
              type="text"
              {...field}
            />
            <p className="text-[11px] xs:text-xs text-red-500">
              {facebookErrorMsg}
            </p>
          </>
        )}
      />

      <Controller
        name="instagramUrl"
        control={control}
        rules={{
          required: MESSAGES_ERROR.URL_REQUIRED,
          pattern: {
            value: REGEX.URL,
            message: MESSAGES_ERROR.URL_REQUIRED,
          },
        }}
        render={({ field }) => (
          <>
            <InputField
              label="Instagram Account"
              placeholder=""
              type="text"
              {...field}
            />
            <p className="text-[11px] xs:text-xs text-red-500">
              {instagramErrorMsg}
            </p>
          </>
        )}
      />
    </div>
  );
};

export default Socials;
