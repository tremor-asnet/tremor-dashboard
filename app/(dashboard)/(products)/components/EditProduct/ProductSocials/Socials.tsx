"use client";

import { Controller, useFormContext } from "react-hook-form";

// Components
import { Title, Flex, Card } from "@tremor/react";

//Styles
import "@/styles/products.css";

const Socials = () => {
  const { control } = useFormContext();

  return (
    <Card className="dark:bg-dark-tremor-primary ring-0">
      <Title className="font-primary font-bold text-primary dark:text-dark-primary text-xl leading-snug capitalize">
        Socials
      </Title>
      <Flex className="flex-col items-start justify-start mt-8">
        <Controller
          control={control}
          render={({ field }) => (
            <input
              className="input-text"
              id="shoppify-handle"
              placeholder="Shoppify Handle"
              {...field}
            />
          )}
          name="shopifyUrl"
        />
        <Controller
          control={control}
          render={({ field }) => (
            <input
              className="input-text"
              id="facebook-account"
              placeholder="Facebook Account"
              {...field}
            />
          )}
          name="facebookUrl"
        />
        <Controller
          control={control}
          render={({ field }) => (
            <input
              className="input-text"
              id="instagram-account"
              placeholder="Instagram Account"
              {...field}
            />
          )}
          name="instagramUrl"
        />
      </Flex>
    </Card>
  );
};

export default Socials;
