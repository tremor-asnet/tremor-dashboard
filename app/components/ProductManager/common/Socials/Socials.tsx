// Libs
import { Control, Controller } from "react-hook-form";

// Components
import InputField from "@/components/common/CustomField/InputField/InputField";

// Types
import { NewSocial } from "@/types";

interface SocialsProps {
  control: Control<NewSocial>;
}

const Socials = ({ control }: SocialsProps) => (
  <div className="flex flex-col gap-6">
    <Controller
      name="shopifyUrl"
      control={control}
      render={({ field }) => (
        <InputField
          label="Shopify Handle"
          placeholder=""
          type="text"
          {...field}
        />
      )}
    />

    <Controller
      name="facebookUrl"
      control={control}
      render={({ field }) => (
        <InputField
          label="Facebook Account"
          placeholder=""
          type="text"
          {...field}
        />
      )}
    />

    <Controller
      name="instagramUrl"
      control={control}
      render={({ field }) => (
        <InputField
          label="Instagram Account"
          placeholder=""
          type="text"
          {...field}
        />
      )}
    />
  </div>
);

export default Socials;
