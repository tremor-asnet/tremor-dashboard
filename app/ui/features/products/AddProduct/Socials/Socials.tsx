// Libs
import { Controller, FieldErrors } from "react-hook-form";

// Components
import { InputField } from "@/ui/components";
import { Text } from "@tremor/react";

// Constants
import { MESSAGES_ERROR, REGEX } from "@/constants";

// Types
import { NewSocial } from "@/types";

interface SocialsProps {
  control: any;
  errors: FieldErrors<NewSocial>;
}

const Socials = ({ control, errors }: SocialsProps) => {
  const facebookErrorMsg = errors.facebookUrl?.message;
  const instagramErrorMsg = errors.instagramUrl?.message;
  const shopifyErrorMsg = errors.shopifyUrl?.message;

  return (
    <>
      <Controller
        name="shopifyUrl"
        control={control}
        rules={{
          required: MESSAGES_ERROR.URL_REQUIRED,
          pattern: {
            value: REGEX.URL,
            message: MESSAGES_ERROR.URL_REQUIRED,
          },
        }}
        render={({ field }) => (
          <div className="mb-3">
            <InputField
              id="add-product-shopify"
              label="Shopify Handle"
              type="text"
              {...field}
            />
            <Text className="text-xs text-red-500 dark:text-red-500 mt-1">
              {shopifyErrorMsg}
            </Text>
          </div>
        )}
      />

      <Controller
        name="facebookUrl"
        control={control}
        rules={{
          required: MESSAGES_ERROR.URL_REQUIRED,
          pattern: {
            value: REGEX.URL,
            message: MESSAGES_ERROR.URL_REQUIRED,
          },
        }}
        render={({ field }) => (
          <div className="mb-3">
            <InputField
              id="add-product-fb"
              label="Facebook Account"
              type="text"
              {...field}
            />
            <Text className="text-xs text-red-500 dark:text-red-500 mt-1">
              {facebookErrorMsg}
            </Text>
          </div>
        )}
      />

      <Controller
        name="instagramUrl"
        control={control}
        rules={{
          required: MESSAGES_ERROR.URL_REQUIRED,
          pattern: {
            value: REGEX.URL,
            message: MESSAGES_ERROR.URL_REQUIRED,
          },
        }}
        render={({ field }) => (
          <div className="mb-3">
            <InputField
              id="add-product-ig"
              label="Instagram Account"
              type="text"
              {...field}
            />
            <Text className="text-xs text-red-500 dark:text-red-500 mt-1">
              {instagramErrorMsg}
            </Text>
          </div>
        )}
      />
    </>
  );
};

export default Socials;
