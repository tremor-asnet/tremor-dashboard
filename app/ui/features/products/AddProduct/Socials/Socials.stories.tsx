import type { Meta, StoryObj } from "@storybook/react";

// Libs
import { useForm } from "react-hook-form";

// Components
import Socials from "./Socials";

// Types
import { NewSocial } from "@/types";

const meta = {
  title: "Components/Products/AddProduct/Socials",
  component: Socials,
  tags: ["autodocs"],
  argTypes: {
    control: {
      description:
        "This is control from useForm of react-hook-form with type NewSocial(shopifyUrl: string, facebookUrl: string, instagramUrl: string)",
    },
    errors: {
      description:
        "This is errors from FieldErrors of react-hook-form with type NewSocial()",
    },
  },
} as Meta<typeof Socials>;

export default meta;

type Story = StoryObj<typeof meta>;

const SocialsFormProvider = () => {
  const { control } = useForm<NewSocial>({
    defaultValues: {
      shopifyUrl: "https://shoppify-url.com",
      facebookUrl: "https://facebook-url.com",
      instagramUrl: "https://instagra-url.com",
    },
  });

  return <Socials control={control} />;
};

export const Default: Story = {
  render: () => <SocialsFormProvider />,
};
