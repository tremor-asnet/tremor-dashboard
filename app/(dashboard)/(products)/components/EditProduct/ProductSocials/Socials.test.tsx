import { render } from "@testing-library/react";

// Components
import Socials from "./Socials";

describe("Testing Order detail contact section", () => {
  // TODO: Update unit test later
  it.skip("Should match snapshot", () => {
    const component = render(<Socials />);
    expect(component).toMatchSnapshot();
  });
});
