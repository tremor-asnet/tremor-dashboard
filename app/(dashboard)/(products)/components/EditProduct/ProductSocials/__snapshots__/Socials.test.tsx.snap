// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Order detail contact section Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="tremor-Card-root relative w-full text-left rounded-tremor-default bg-tremor-background ring-tremor-ring shadow-tremor-card dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card border-tremor-brand dark:border-dark-tremor-brand p-6 dark:bg-dark-tremor-primary ring-0"
      >
        <p
          class="text-tremor-title font-primary font-bold text-primary dark:text-dark-primary text-xl leading-snug capitalize"
        >
          Socials
        </p>
        <div
          class="tremor-Flex-root flex w-full flex-col items-start justify-start mt-8"
        >
          <input
            class="input-text"
            id="shoppify-handle"
            placeholder="Shoppify Handle"
            value=""
          />
          <input
            class="input-text"
            id="facebook-account"
            placeholder="Facebook Account"
            value=""
          />
          <input
            class="input-text"
            id="instagram-account"
            placeholder="Instagram Account"
            value=""
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="tremor-Card-root relative w-full text-left rounded-tremor-default bg-tremor-background ring-tremor-ring shadow-tremor-card dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card border-tremor-brand dark:border-dark-tremor-brand p-6 dark:bg-dark-tremor-primary ring-0"
    >
      <p
        class="text-tremor-title font-primary font-bold text-primary dark:text-dark-primary text-xl leading-snug capitalize"
      >
        Socials
      </p>
      <div
        class="tremor-Flex-root flex w-full flex-col items-start justify-start mt-8"
      >
        <input
          class="input-text"
          id="shoppify-handle"
          placeholder="Shoppify Handle"
          value=""
        />
        <input
          class="input-text"
          id="facebook-account"
          placeholder="Facebook Account"
          value=""
        />
        <input
          class="input-text"
          id="instagram-account"
          placeholder="Instagram Account"
          value=""
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
